### PR TITLE
Fixes 2937:   List packages in a snapshot

### PR DESCRIPTION
**Pull request by kwarnerredhat**
## Summary
List the packages in a snapshot

## Testing steps
-click packages in view all snapshots   
-opens new modal  
-goes back to original repo

### CODE REVIEW

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ const perPageKey = 'snapshotPerPage';
 export default function SnapshotListModal() {
   const classes = useStyles();
   const rootPath = useRootPath();
-  const { repoUUID: uuid = '' } = useParams();
+  const { repoUUID: uuid, snapshotUUID = '' } = useParams();
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:70]: as mentioned, no need to add snapshotUUID here as it won't be accessible on [this](https://github.com/content-services/content-sources-frontend/pull/207/files#diff-21c0871aec2cc959996fa3600d513fccb3575d5d72dd892b0102df47b62058acR44) route.

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 2]
```diff
@@ -99,7 +99,7 @@ export default function SnapshotListModal() {
     data = { data: [], meta: { count: 0, limit: 20, offset: 0 } },
   } = useGetSnapshotList(uuid as string, page, perPage, debouncedSearchQuery, sortString);
 
-  const { data: contentData } = useFetchContent([uuid]);
+  const { data: contentData } = useFetchContent([snapshotUUID]);
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:102]: Same as above, no need to change this value here, uuid in this case is actually repoUUID, we want to make sure we fetch the correct repo, a snapshot uuid here wouldn't work.

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 3]
```diff
@@ -228,28 +228,41 @@ export default function SnapshotListModal() {
...
+                            />
+                          </Link>
+                        )}
                       </Td>
                       <Td>{content_counts?.['rpm.package'] || 0}</Td>
                       <Td>{content_counts?.['rpm.advisory'] || 0}</Td>
                       <Td>
-                        <RepoConfig repoUUID={uuid} snapUUID={snap_uuid}/>
-                        </Td>
+                        <RepoConfig uuid={uuid} snapUUID={snap_uuid} />
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:264]: no need to change the variables on this component, as mentioned on our call, you can revert this change.

File: src/Pages/ContentListTable/components/SnapshotListModal/components/RepoConfig.tsx [hunk 4]
```diff
@@ -15,12 +15,12 @@ const useStyles = createUseStyles({
   },
 });
 
-interface Props {
+interface RepoConfigProps {
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/components/RepoConfig.tsx:18]: you can revert all changes in this file.

File: src/Routes/useTabbedRoutes.tsx [hunk 5]
```diff
@@ -39,7 +40,10 @@ export default function useTabbedRoutes(): TabbedRoute[] {
               ]
             : []),
           ...(features?.admintasks?.enabled && features.snapshots?.accessible
-            ? [{ path: ':repoUUID/snapshots', Element: SnapshotListModal }]
+            ? [
+                { path: ':repoUUID/snapshots', Element: SnapshotListModal },
+                { path: ':repoUUID/snapshots/:snapshotUUID', Element: SnapshotPackageModal },
```
Comment (Andrewgdewar) [inline src/Routes/useTabbedRoutes.tsx:45]: 👍

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 6]
```diff
@@ -144,6 +145,14 @@ export default function SnapshotListModal() {
 
   const loadingOrZeroCount = fetchingOrLoading || !count;
 
+  const onPackageCountClick = (snapUUID) => {
+    const SnapshotListModalRoute = `/content/${uuid}/snapshots/${snapUUID}/packages`;
+
+    navigate(
+      `${window.location.pathname}?modalRoute=${encodeURIComponent(SnapshotListModalRoute)}`,
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:152]: not sure where you found this, but we have [a different pattern](https://github.com/Andrewgdewar/content-sources-frontend/blob/fe21d61edd93c360cc74026642aa8ba58baa6ad5/src/Pages/AdminTaskTable/components/ViewPayloadModal/ViewPayloadModal.tsx#L40) to use for routing.

Comment (Andrewgdewar) [reply src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:152]: Look at the linked pattern above, use a combination of that and the snapshotlistmodalroute youv'e created above to get the navigate opening your modal.

Comment (Andrewgdewar) [reply src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:152]: Hint: [This](https://github.com/content-services/content-sources-frontend/pull/207/files#diff-21c0871aec2cc959996fa3600d513fccb3575d5d72dd892b0102df47b62058acR45) is the route that you are trying to go to.

Comment (Andrewgdewar) [reply src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:152]: Hint: [This](https://github.com/content-services/content-sources-frontend/pull/207/files#diff-21c0871aec2cc959996fa3600d513fccb3575d5d72dd892b0102df47b62058acR44) is where you are when you click the link.

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-2937

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 7]
```diff
@@ -228,28 +234,42 @@ export default function SnapshotListModal() {
...
-                            (added_counts?.['rpm.advisory'] || 0) +
-                            (added_counts?.['rpm.package'] || 0)
-                          }
-                          removedCount={
-                            (removed_counts?.['rpm.advisory'] || 0) +
-                            (removed_counts?.['rpm.package'] || 0)
-                          }
-                        />
+                        {snap_uuid && (
+                          <Link to={`${rootPath}${SnapshotListModalRoute(snap_uuid)}`}>
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:250]: You are using a react-router link component here. 
Although this may work, I think as you've created functions above, you'll likely want to use a button to encapsulate the changed arrows with an [inline link](https://www.patternfly.org/components/button) button, as seen on the Patternfly component page. 
You can then use the regular "onClick" or the "href" variable depending.

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 8]
```diff
@@ -143,7 +143,13 @@ export default function SnapshotListModal() {
   const fetchingOrLoading = isFetching || isLoading;
 
   const loadingOrZeroCount = fetchingOrLoading || !count;
+  const SnapshotListModalRoute = (snapUUID) => `/content/${uuid}/snapshots/${snapUUID}/packages`;
 
+  const onPackageCountClick = (snapUUID) => {
+    const SnapshotListModalRoute = `/content/${uuid}/snapshots/${snapUUID}/packages`;
+
+    navigate(`${rootPath}${SnapshotListModalRoute}`);
+  };
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:152]: You have two functions doing the same thing, and the routes aren't correct on either one. 
The[ route you created](https://github.com/content-services/content-sources-frontend/pull/207/files#diff-21c0871aec2cc959996fa3600d513fccb3575d5d72dd892b0102df47b62058acR45) below does not match.

Choose one of the above functions/methods to navigate.

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 9]
```diff
@@ -228,28 +234,42 @@ export default function SnapshotListModal() {
...
-                          addedCount={
-                            (added_counts?.['rpm.advisory'] || 0) +
-                            (added_counts?.['rpm.package'] || 0)
-                          }
-                          removedCount={
-                            (removed_counts?.['rpm.advisory'] || 0) +
-                            (removed_counts?.['rpm.package'] || 0)
-                          }
-                        />
+                        {snap_uuid && (
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:249]: You don't need to conditionally show this section. 
And if you do, do not use  `{snap_uuid &&`
[Read this](https://medium.com/@davidkelley87/avoiding-in-reacts-conditional-rendering-dec9eb61ee65#:~:text=Using%20%E2%80%9C%26%26%E2%80%9D%20to%20conditionally%20render%20components%20can%20lead%20to%20unexpected,evaluate%20to%20the%20expected%20value.)

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 10]
```diff
@@ -245,11 +251,25 @@ export default function SnapshotListModal() {
...
+                        <Button
+                          onClick={() => {
+                            navigate(`${rootPath}/${uuid}/snapshots/${snap_uuid}`);
+                          }}
+                          style={{
+                            color: 'lightblue',
+                            border: 'none',
+                            background: 'none',
+                            padding: 0,
+                          }}
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:264]: You don'e need to create a custom style here (or shouldn't at least). 
You can look at the 'Inline link' variant found [here](https://www.patternfly.org/components/button#variant-examples).

**Comment by Andrewgdewar** [comment]
Screenshot 2024-02-15 at 10 47 06 AM

Look at the design for the title of he modal, you need to update this.

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+    <Modal
+      key={uuid}
+      position='top'
+      hasNoBodyWrapper
+      aria-label='RPM package modal'
+      ouiaId='rpm_package_modal'
+      ouiaSafe={fetchingOrLoading}
+      variant={ModalVariant.medium}
+      title='Snapshot Details'
+      description={<p className={classes.description}>Here are the snapshot details below</p>}
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx:152]: Maybe just comment this line out for now. As we don't have copy (text) in the design for this yet.

File: src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx [hunk 12]
```diff
@@ -245,11 +251,19 @@ export default function SnapshotListModal() {
                           }
                         />
                       </Td>
-                      <Td>{content_counts?.['rpm.package'] || 0}</Td>
+                      <Td>
+                        <Button
+                          onClick={() => {
+                            navigate(`${rootPath}/${uuid}/snapshots/${snap_uuid}`);
+                          }}
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:258]: Read my previous comment, you need to add the variant to this button as described there.

Comment (Andrewgdewar) [reply src/Pages/ContentListTable/components/SnapshotListModal/SnapshotListModal.tsx:258]: Screenshot 2024-02-15 at 2 10 37 PM
You need to remember to test your changes before committing.

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx (hunk 11 above)
Comment (xbhouse) [reply src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx:152]: also small thing, but `Snapshot Details` should be `Snapshot detail` i think so we don't get yelled at again for not using sentence case :)

Comment (kwarnerredhat) [reply src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx:152]: On it

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx [hunk 13]
```diff
@@ -76,3 +76,30 @@ it('Render with no packages (after an unsuccessful search)', () => {
...
+  }));
+
+  const { getByText, findByText } = render(
+    <ReactQueryTestWrapper>
+      <SnapshotPackageModal />
+    </ReactQueryTestWrapper>,
+  );
+
+  fireEvent.click(getByText('Packages'));
+  expect(await findByText('View All Snapshots')).toBeInTheDocument();
```
Comment (jlsherrill) [inline src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx:96]: I'm a bit confused what this is doing.  You are rendering the modal, and then trying to find "Packages" which isn't in the modal and click on it?

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+  const fetchingOrLoading = isFetching || isLoading;
+
+  const loadingOrZeroCount = fetchingOrLoading || !count;
+
+  return (
+    <Modal
+      key={uuid}
+      position='top'
+      hasNoBodyWrapper
+      aria-label='RPM package modal'
```
Comment (xbhouse) [inline src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx:147]: i think you'll want to change this aria-label and ouiaId from RPM package modal to snapshot package modal to differentiate between the 2 modals

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+  expect(queryByText('Packages')).toBeInTheDocument();
+  expect(queryByText(packageItem.name)).toBeInTheDocument();
+  expect(queryByText(packageItem.version)).toBeInTheDocument();
+  expect(queryByText(packageItem.release)).toBeInTheDocument();
+  expect(queryByText(packageItem.arch)).toBeInTheDocument();
+  expect(queryByText('Clear search')).not.toBeInTheDocument();
+});
+
+it('Render with no packages (after an unsuccessful search)', () => {
```
Comment (xbhouse) [inline src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx:60]: i'm not sure this test case makes sense yet since i don't think you've implemented the search within the SnapshotPackageModal, right?

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx (hunk 13 above)
Comment (jlsherrill) [reply src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx:96]: oh, i see, i think you are wanting to:

* Load the repositories list
* click view all snapshots
* click on the package count (opens the modal)
* click on something to close the modal
* verify you are back on the correct page.  

Is that right?

Comment (kwarnerredhat) [reply src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx:96]: Yeah

Comment (kwarnerredhat) [reply src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx:96]: Thats right

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx [hunk 16]
```diff
@@ -144,8 +144,8 @@ export default function SnapshotPackageModal() {
       key={uuid}
       position='top'
       hasNoBodyWrapper
-      aria-label='RPM package modal'
-      ouiaId='rpm_package_modal'
+      aria-label='snapshot package modal'
+      ouiaId='snapshot package modal'
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx:148]: ouiaId's usually use snake_case, fix this please!

File: src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+//     },
+//   }));
+
+//    const { getByText, queryByText } = render(
+//     <ReactQueryTestWrapper>
+//       <SnapshotPackageModal />
+//      </ReactQueryTestWrapper>,
+//   );
+
+// });
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.test.tsx:77]: delete this if it's not used.

**Comment by Andrewgdewar** [comment]
This is looking good, but I'm going to close this for now and open another PR with a few of the other components of this modal added in.
